arelle/UrlUtil.py

```diff
@@ -38,21 +41,21 @@ def scheme(url: str) -> str | None: # returns None if no scheme part
 # This regular expression is only partial validation.
 relativeUrlPattern = re.compile(r"^(urn:|(([a-zA-Z][a-zA-Z0-9.+-]+):)?(//([^/\?#]*))?(?![^:/]*:[^/]*(/|$)))([^\?#]*)(\?([^#]*))?(#([^#]*))?$")
 
-def splitDecodeFragment(url):
+def splitDecodeFragment(url: str) -> tuple[str, str]:
     if url is None: # urldefrag returns byte strings for none, instead of unicode strings
-        return _STR_UNICODE(""), _STR_UNICODE("")
+        return _STR_UNICODE(""), _STR_UNICODE("") # type: ignore[name-defined]
```

Comment:
Not required, but many of these [undefined built-ins](https://github.com/Arelle/Arelle/blob/d9fe63173b84df74d3ed15a0afe37214df85ca82/arelle/PythonUtil.py#L16) are coming from python 2/3 handling. If you want, instead of `ignore[name-defined]` you can swap these out with the python 3 function, or in this case remove the function that's redundant in python 3 and return literal empty strings instead.

Reply:
Oh, so this is basically just `return ""`?

Reply:
It is returning a tuple of two empty strings, but yeah, that's all it's doing (in python 3). It was required for python 2 unicode handeling, but in python 3 it's just calling `str()` and `str("") == """`  so it's identical to:
```suggestion
        return "", ""
```

Reply:
I'll keep this in mind going forward